### PR TITLE
Fix prebuilt package resolution in the new pex when pexs have '#' in the name.

### PR DESCRIPTION
**Pull request by mikekap**
Buck-based python tests tend to have these.

### CODE REVIEW

File: third-party/py/pex/pex/link.py [hunk 1]
```diff
@@ -73,7 +73,9 @@ def __init__(self, url):
     """
     purl = urlparse.urlparse(url)
     if purl.scheme == '':
-      purl = urlparse.urlparse(self._normalize(url))
+      purl = urlparse.urlparse(self._normalize(url), allow_fragments=False)
+    elif purl.scheme == 'file':
+      purl = urlparse.urlparse(url, allow_fragments=False)
```
Comment (sdwilsh) [inline third-party/py/pex/pex/link.py]: This also adds support for file, right?  Care to update the message and `README.facebook` to be more clear?

Comment (mikekap) [inline third-party/py/pex/pex/link.py]: file:// worked previously - see _normalize() - it just adds a file:// prefix. This just makes sure fragments are ignored when there's no scheme or the scheme is file://

Comment (sdwilsh) [inline third-party/py/pex/pex/link.py]: Yup.  I'm just blind.  I think I should stop reviewing code today :)

**Comment by sdwilsh** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**